runner/utils/driver/istio/names.go

```diff
@@ -27,10 +28,42 @@ func getInternalServiceName(serviceID, version string) string {
 	return fmt.Sprintf("%s-%s-internal", serviceID, version)
 }
 
+func splitInternalServiceName(n string) (serviceID, version string) {
+	arr := strings.Split(n, "-")
+	if len(arr) < 3 || arr[len(arr)-1] != "internal" {
+		return "", ""
+	}
+	return strings.Join(arr[:len(arr)-2], "-"), arr[len(arr)-2]
+}
+
 func getInternalServiceDomain(projectID, serviceID, version string) string {
 	return fmt.Sprintf("%s.%s.svc.cluster.local", getInternalServiceName(serviceID, version), projectID)
 }
 
+func checkIfInternalServiceDomain(projectID, serviceID, internalServiceDomain string) bool {
+	arr := strings.Split(internalServiceDomain, ".")
+	if strings.HasSuffix(internalServiceDomain, "svc.cluster.local") {
+		serID, _ := splitInternalServiceName(arr[0])
+		if len(arr) == 5 && arr[1] == projectID && serID == serviceID {
+			return true
+		}
+	}
+	return false
+}
+
+func splitInternalServiceDomain(s string) (projectID, serviceID, version string) {
```

Comment:
I don't think we need this function. Make the check function return everything along with thr Boolean